torch/csrc/autograd/python_variable.cpp

```diff
@@ -105,6 +105,44 @@ class PyInterpreterHolder {
 };
 PyInterpreterHolder self_interpreter;
 
+// Utility function that can be used to properly clear a pyobj associated
+// with the given Tensor.
+// For classic binding, this would be as simple as clearing the field and
+// decrefing the pyobj but the special logic that keep Tensor's pyobj alive
+// would prevent this from working as expected.
+// Here we first properly break the link Tensor -> pyobj. Then we break the
+// link pyobj -> Tensor. And finally decref the pyobj.
+//
+// This function assumes that it is used just before calling into
+// init_pyobj() (via THPVariable_NewWithVar) in particular for the interpreter
+// state.
+static void unsafe_clear_pyobj(const Variable& tensor) {
```

Comment:
I object to this function existing. It breaks the invariant that once a C++ tensor is associated with a PyObject, it is always associated with that PyObject

Reply:
I agree with that.
I wanted to have a full fix here to be able to drive the discussion.